pkg/storage/podmetrics/reststorage.go

```diff
@@ -84,6 +87,16 @@ func (m *MetricStorage) List(ctx context.Context, options *metainternalversion.L
 		return &metrics.PodMetricsList{}, errMsg
 	}
 
+	// currently the PodLister API does not support filtering using FieldSelectors, we have to filter manually
+	if options != nil && options.FieldSelector != nil {
+		for i := len(pods) - 1; i >= 0; i-- {
+			fieldsSet := generic.AddObjectMetaFieldsSet(make(fields.Set, 2), &pods[i].ObjectMeta, true)
+			if options.FieldSelector.Matches(fieldsSet) {
+				pods = append(pods[:i], pods[i+1:]...)
```

Comment:
I may be missing something, but why are we _removing_ the pod if the field selector matches?

Reply:
nope it's a bug, I must have been drunk when I was testing it. I better added some unit tests to the PR. Thanks for catching

Reply:
@11janci no worries and thanks a lot for the contribution! Indeed, do you mind to add some unit tests here?

Reply:
already done, pls refresh the PR. Thanks again